doc/user/metrics/README.md

```diff
@@ -74,15 +74,129 @@ In Kubernetes clusters where [OwnerReferencesPermissionEnforcement][ownerref-per
   - update
 ...
 ```
-
 ### Custom resource specific metrics
 
 By default operator will expose info metrics based on the number of the current instances of an operator's custom resources in the cluster. It leverages [kube-state-metrics][ksm] as a library to generate those metrics. Metrics initialization lives in the `cmd/manager/main.go` file of the operator in the `serveCRMetrics` function. Its arguments are a custom resource's group, version, and kind to generate the metrics. The metrics are served on `0.0.0.0:8686/metrics` by default. To modify the exposed metrics port number, change the `operatorMetricsPort` variable at the top of the `cmd/manager/main.go` file in the generated operator.
 
+### Custom Metrics using Operator-SDK
+To implement custom metric collection using the `metrics package` provided by SDK, follow the steps outlined below:
+1. Update the Operator: Update the operator to provide sufficient permisisons so that the required resources for prometheus can be created.
+2. Custom metric collection: Implement the logic for custom metric variables defined for your operator and register the specific metrics with prometheus. 
+3. Expose metrics: Utilize the functions provided by SDK to expose metrics and start the metrics server.
+
+#### Update the Operator
+
+Update the operator and provide sufficient permissions to allow creation of Prometheus resources. The required permissions are:
+1. `Service Monitor` : `get` and `create`
+2. `roles` and `rolebindings` : `get`, `create`, `delete`, `update`, `list`
+
+The below is a snippet from roles.yaml:
+```yaml
+- apiGroups:
+  - rbac.authorization.k8s.io
+  resources:
+  - roles
+  - rolebindings
+  verbs:
+  - get
+  - create
+  - delete
+  - update
+  - list
+  - watch
+- apiGroups:
+  - monitoring.coreos.com
+  resources:
+  - servicemonitors
+  verbs:
+  - get
+  - create
+
+```
+#### Custom metric collection
+
+Implement the logic to collect specific custom metrics inside the operator code base. For example, the following is a snippet from `Marketplace` operator which implements custom metric collection:
```

Comment:
```suggestion
Following an example of business logic implementation for customized metrics.  
```

Reply:
Could you please add an example based on the Memcached sample? 

Also, WDYT to add a NOTE with the link for the code source where has this impl in the `Marketplace` project as the following example?

 ```suggestion
...
**NOTE:** To check a further example see it implemented [here][] in the [Marketplace][marketplace] project  
```

Reply:
Im not sure on whether the metrics collection is completely done on marketplace. I have provided a CounterVec as an example. Users have directly use this inside their operator. I'm not sure on what is required with specific to Memcached sample. Could you please elaborate more on that?

Reply:
But you added `following is a snippet from `Marketplace`, so would be great to have the link for this piece of code there and the `Marketplace`. Or, it should not be referenced.  is it make sense?